freeipa/src/main/java/com/sequenceiq/freeipa/service/image/ImageCatalogProvider.java

```diff
@@ -88,7 +91,8 @@ public void evictImageCatalogCache(String catalogUrl) {
 
     private ImageCatalog filterImagesByOsType(ImageCatalog catalog) {
         LOGGER.debug("Filtering images by OS type {}", getEnabledLinuxTypes());
-        if (CollectionUtils.isEmpty(getEnabledLinuxTypes()) || Objects.isNull(catalog) || Objects.isNull(catalog.getImages())) {
+        if ((CollectionUtils.isEmpty(getEnabledLinuxTypes()) || Objects.isNull(catalog) || Objects.isNull(catalog.getImages()))
+                && catalog.getVersions() != null) {
```

Comment:
If the catalog is null, this will try to evaluate if its versions are not null, resulting in NPE

Reply:
You are right. We should refactor this as the original implementation is totally a mess. It can simply result in a null image catalog what will cause an NPE later...